### DISCUSSION TITLE
database notifications conditionally hide mark as unread/read button

### DISCUSSION THREAD

**Discussion by juliangums** [filamentphp/filament#12527] [Help]
### Package

Notifications

### Package Version

latest

### How can we help you?

I can add those buttons to a notification, but I only want it to be rendered if it is actually needed. If the notification is read. I don't know how I can inject a dependency in the closure, so I can conditionally hide it. I tried $record, $value, $notification, $livewire but the only thing that worked was $action and from there I don't know how to get the notification. Any ideas?

```php
            ->actions([
                Action::make('markAsRead')
                    ->button()
                    ->markAsRead(),
                Action::make('markAsUnread')
                    ->button()
                    ->markAsUnread(),
            ])
```

**Comment by juliangums** [comment]
I still haven't found a solution for this. Anyone?

**Reply by leandrocfe** [reply]
I'll try this and I let you know..

**Reply by juliangums** [reply]
@leandrocfe have you had a chance to try this?

**Reply by leandrocfe** [reply]
I forgot it Lol

**Comment by FrazeColder** [comment]
Any news here?

**Comment by sohailahmad07** [comment]
Did you guys find a solution to this

**Reply by juliangums** [reply]
Unfortunately not. Feel free to submit a PR that extends those injections

**Comment by dxpawan** [comment]
We can add class to mark as read button and hide/show with css


->actions([
                Action::make('markAsRead')
                    ->label('Mark as Read')
                    ->markAsRead()
                    ->extraAttributes(['class' => 'mark-as-read-button'])
            ])


.mark-as-read-button{
    display: none !important;
}
.before\:bg-primary-600 .mark-as-read-button {
    display: inline-block !important;
}

**Reply by juliangums** [reply]
Thanks. Not really a great solution, basing this on CSS

**Comment by thaqebon** [comment]
or just tailwind classes

```php
Action::make('dismiss')
    ->extraAttributes(['class' => '[.fi-no-notification-read-ctn_&]:hidden'])
    ->markAsRead(),

Action::make('markAsUnread')
    ->extraAttributes(['class' => '[.fi-no-notification-unread-ctn_&]:hidden'])
    ->markAsRead(),
```
